runtime/lua/vim/filetype.lua

```diff
@@ -466,6 +466,7 @@ local extension = {
   glsl = 'glsl',
   gn = 'gn',
   gni = 'gn',
+  gnuplot = 'gnuplot',
```

Comment:
Also need to remove line 1308

Reply:
No.

Reply:
Why? It's duplicate.

Reply:
No it's not. Test fails if I remove it.

Reply:
Oh, that seems to be a much bigger problem.

Reply:
No, just a standard issue that `extension` doesn't apply to `.extension` files?

Reply:
This means Nvim's behavior is different from Vim for many `.extension` files.

Reply:
Where this matters, we have added the `.extension` files separately. That's the whole point of the filetype tests.

Reply:
> Where this matters

Where does this not matter? It should matter for any extension.

Reply:
In practice. Nobody (sane) is working with literal `.tex` files.

Reply:
Yes, but that still matters, doesn't it? I think this needs more discussion in #27976.